src/Schema/Directives/Nodes/UnionDirective.php

```diff
@@ -46,10 +42,26 @@ function() use ($value){
                     return resolve(TypeRegistry::class)->get($type->name->value);
                 })->filter()->toArray();
             },
+            'resolveType' => $this->getResolver(function ($value, Context $context, ResolveInfo $info) {
+                $unionName = $this->definitionNode->name->value;
 
-            'resolveType' => function ($value) use ($resolver) {
-                return $resolver($value);
-            },
+                // Try to locate a fallback resolver corresponds to the config file.
+                $resolverClass = config('lighthouse.namespaces.unions') . '\\' . $unionName;
+
+                if (\class_exists($resolverClass)) {
+                    $resolver = resolve($resolverClass);
+
+                    if ( ! \method_exists($resolver, 'resolve')) {
+                        throw new DirectiveException("Class {$resolverClass} must have a `resolve` method.");
+                    }
+
+                    return $resolver->resolve($value, $context, $info);
+                }
+
+                return resolve(TypeRegistry::class)->get(
+                    str_after('\\', \get_class($value))
```

Comment:
I just noticed that this is a pretty severe bug (that is still in the previous version, it seems). Apparently, no one actually uses Union types 🙊 

Anyways, can you please provide a test that shows that this works, and fix this one?

Reply:
@spawnia 

What's wrong with this one…?
Do you mean If the $value is not an object get_class($value) won't get any class?

Reply:
The order of the arguments in `str_after` is reversed

Reply:
Got that!